extensions/google-java/src/main/kotlin/com/connectrpc/extensions/GoogleJavaJSONStrategy.kt

```diff
@@ -18,25 +18,31 @@ import com.connectrpc.CODEC_NAME_JSON
 import com.connectrpc.Codec
 import com.connectrpc.ErrorDetailParser
 import com.connectrpc.SerializationStrategy
-import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Message
+import com.google.protobuf.TypeRegistry
 import kotlin.reflect.KClass
+import kotlin.reflect.full.isSubclassOf
 
 /**
  * The Google Java JSON serialization strategy.
  */
-class GoogleJavaJSONStrategy : SerializationStrategy {
+class GoogleJavaJSONStrategy(
+    private val registry: TypeRegistry = TypeRegistry.getEmptyTypeRegistry(),
+) : SerializationStrategy {
     override fun serializationName(): String {
         return CODEC_NAME_JSON
     }
 
-    /**
-     * This unchecked cast assumes the underlying class type is
-     * a Google GeneratedMessageV3.
-     */
-    @Suppress("UNCHECKED_CAST")
     override fun <E : Any> codec(clazz: KClass<E>): Codec<E> {
-        val messageClass = clazz as KClass<GeneratedMessageV3>
-        return GoogleJavaJSONAdapter(messageClass) as Codec<E>
+        if (!clazz.isSubclassOf(Message::class)) {
```

Comment:
Should this be `MessageLite`?

Reply:
Unfortunately, no. The JSON format requires full messages since it must examine the descriptor to discover the `json_name` option.

Reply:
Ok then we should update the message on line 38 to use Message instead of MessageLite in the exception.

Reply:
Ah, right! Thanks for catching that!

Fixed. I also added a few TODOs based on other comments I left here in a self-review, since in the code seems better than in a PR comment.